artifacts/definitions/Windows/Forensics/BulkExtractor.yaml

```diff
@@ -0,0 +1,163 @@
+    description: "Bulk_extractor custom commands. .e.g '-E evtx, -e zip -S unzip_carve_mode=2'"
+   
+sources:
+  - queries:
+    - LET bin <= SELECT * 
+        FROM Artifact.Generic.Utils.FetchBinary(
+            binaryURL=binaryURL, ToolName="Bulk_Extractor")
+    - LET tempfolder <= tempdir
+    - LET zip_file <= SELECT 
+            copy(filename=FullPath, dest=tempfile(extension=".exe"), 
+                accessor='zip') AS Payload 
+        FROM glob(globs="file:///" + (bin[0]).FullPath + 
+             "#/bulk_extractor.exe", accessor='zip')
+    - LET target = SELECT 
+            DeviceID,
+            if(condition=DeviceID=~"^\\\\\\\\.\\\\",
+                then=split(string=split(string=DeviceID,sep='\\\\\\\\.\\\\')[1],
+                    sep='\\\\')[0],
+                else="bulk_out") as base,
+            _DeviceID
+        FROM chain(
+            a={
+                SELECT 
+                    DeviceID,
+                    upcase(string=DeviceID) as _DeviceID
+                FROM Artifact.Windows.Sys.DiskInfo()
+                WHERE TargetAllPhysical
+            },
+            b={
+                SELECT 
+                    Target as DeviceID,
+                    upcase(string=Target) as _DeviceID
+                FROM scope()
+                WHERE Target =~ '.'
+            },
+            c={
+                SELECT 
+                    regex_replace(source=FullPath,
+                        re="GLOBALROOT\\\\Device\\\\",replace="")AS DeviceID,
+                    Data.ID AS ShadowCopyID,
+                    upcase(string=regex_replace(source=FullPath,
+                        re="GLOBALROOT\\\\Device\\\\",replace="")) as _DeviceID
+                FROM glob(globs='/*', accessor='ntfs')
+                WHERE ShadowCopyID AND TargetVSS
+                ORDER by FullPath
+            })
+            GROUP BY _DeviceID
+    - LET cmdline = SELECT (zip_file[0].Payload,'-q','99999999999','-R') + CMD + ('-o') as CMD FROM switch(
+            a= { 
+                SELECT split(string=FreeCommand, sep=' ') AS CMD 
+                FROM scope() 
+                WHERE FreeCommand
+            },
+            b= { 
+                SELECT 
+                    ('-E','evtx','-e','find','-f',FindRegex) AS CMD 
+                FROM scope() 
+                WHERE CarveEvtx AND FindRegex
+            }, 
+            c= { 
+                SELECT ('-E','evtx') AS CMD 
+                FROM scope() 
+                WHERE CarveEvtx
+            }, 
+            d= { 
+                SELECT ('-E','find','-f',FindRegex) AS CMD 
+                FROM scope() 
+                WHERE FindRegex
+            }, 
+            e= { 
+                SELECT ('-h') AS CMD FROM scope()
+            })
+    - SELECT * FROM foreach(
+        row=target, 
+        query= { 
+            SELECT * 
+            FROM execve(argv=cmdline[0].CMD + (tempfolder + '\\' + base, DeviceID)
+                , length=10000000)})
```

Comment:
maybe use `sep="\n"` to have  one line per row?

Reply:
I added this in. Im always two minds about this when there is a usecase of multiple executions.